### ISSUE TITLE
Create one more haeder network to estimate distance for each object

### ISSUE THREAD

**Issue by chowkamlee81** [ultralytics/yolov5#2494]
## ❔Question

Hi Glenn,

Would like to estimate distance by creating one more header network. Can you kindly tell how to it since i had groundtruth for distance on mydata.

Requesting you to do the needful.
## Additional context

Hi Glenn,

Would like to estimate distance by creating one more header network. Can you kindly tell how to it since i had groundtruth for distance on mydata.

Requesting you to do the needful.

**Comment by chowkamlee81** [comment]
@glenn-jocher , Kindly suggest and do the needful

**Comment by glenn-jocher** [comment]
@chowkamlee81 if you have additional labels for your data, i.e. shape, color, distance, rotation, etc., you can add these to the dataloader and loss function so the model learns them the same way it learns the current labels. 

The only architecture change you need is to reshape the Detect() layer outputs by adding extra neurons to the current 85 per anchor (for COCO), i.e. 86 per anchor.

https://github.com/ultralytics/yolov5/blob/ed2c74218d6d46605cc5fa68ce9bd6ece213abe4/models/yolo.py#L22-L57

**Comment by chowkamlee81** [comment]
Thanks @glenn-jocher understood. Closing the issue

**Comment by montmejat** [comment]
Hello @chowkamlee81, were you able to implement this? On my side, having a new output and adding a new part in the loss function has a significantly bad impact on the detection precision :confused:
